lua/conform/runner.lua

```diff
@@ -169,6 +169,13 @@ M.apply_format = function(bufnr, original_lines, new_lines, range, only_apply_ra
   table.remove(original_lines)
   table.remove(new_lines)
 
+  -- Abort if output is empty but input is not (i.e. has some non-whitespace characters).
+  -- This is to hack around oddly behaving formatters (e.g black outputs nothing for excluded files).
+  if new_text:match("^%s*$") and not original_text:match("^%s*$") then
+    log.trace("Aborting because a formatter returned empty output")
+    return
+  end
```

Comment:
I think this is a fine place for it, but I'd say add the buffer name to the log line and make it a `warn` instead of a `trace`